src/test/java/org/jitsi/meet/test/web/WebParticipant.java

```diff
@@ -157,6 +159,126 @@ protected void doHangUp()
     }
 
+    /**
+     * Executes a script in this {@link WebParticipant}'s {@link WebDriver}.
+     * See {@link JavascriptExecutor#executeScript(String, Object...)}.
+     */
+    @Override
+    public Object executeScript(String var1, Object... var2)
+    {
+        JavascriptExecutor executor = getJSExecutor();
+
+        return executor != null
+            ? executor.executeScript(var1, var2) : null;
+    }
+
+    /**
+     * Executes a script asynchronously in this {@link WebParticipant}'s
+     * {@link WebDriver}.
+     * See {@link JavascriptExecutor#executeScript(String, Object...)}.
+     */
+    @Override
+    public Object executeAsyncScript(String var1, Object... var2)
+    {
+        JavascriptExecutor executor = getJSExecutor();
+
+        return executor != null
+            ? executor.executeAsyncScript(var1, var2) : null;
+    }
+
+    private JavascriptExecutor getJSExecutor()
+    {
+        return driver instanceof JavascriptExecutor
+            ? (JavascriptExecutor) driver : null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEndpointId()
+    {
+        Object o = executeScript("return APP.conference.getMyUserId();");
+        return o == null ? null : o.toString();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getMeetDebugLog()
+    {
+        try
+        {
+            Object log
+                = executeScript(
+                "try{ "
+                        + "return JSON.stringify("
+                        + "  APP.conference.getLogs(), null, '    ');"
+                        + "}catch (e) {}");
+
+            return log instanceof String ? (String) log : null;
+        }
+        catch (Exception e)
+        {
+            Logger.getGlobal().log(
+                    Level.SEVERE,
+                    "Failed to get meet logs from " + name,
+                    e);
+
+            return null;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getProtocol()
+    {
+        WebDriver driver = getDriver();
+
+        if (driver == null)
+        {
+            return null;
+        }
+        Object protocol
+            = executeScript(
+            "try {"
+                    + "return APP.conference.getStats().transport[0].type;"
+                    + "} catch (err) { return 'error: '+err; }");
+
+        return (protocol == null) ? null : protocol.toString().toLowerCase();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isP2pConnected()
+    {
+        // FIXME hmm this checks for ICE connected state which does not
+        // necessarily mean that the conference is currently in the P2P mode.
+        // It may turn out we're not really checking if P2P is connected
+        // and the method name may be confusing.
+        Object result = executeScript(MeetUtils.ICE_CONNECTED_CHECK_SCRIPT);
+
+        return Boolean.valueOf(result.toString());
```

Comment:
haha, each of these very similar methods is implemented differently (one does a toString, one casts to String, one does a try/catch). :)
This one seems prone to throw a NPE if `result` is null. Just an observation, not a critique of this PR, and I don't expect a fix.

Reply:
I will fix that. Kind of wasn't focused on those details more on moving the code around. Thanks !

Reply:
I'm thinking what's the best way. How about this ?
return result != null && Boolean.valueOf(result.toString());
(possible wrapped into a method)